### PR TITLE
File rights and copy files

### PR DESCRIPTION
**Pull request by spaneni**
- fixed file rights handling
- Added the options "copy-files" to copy files without processing

### CODE REVIEW

File: fp/template-engine.js [hunk 1]
```diff
@@ -169,9 +178,10 @@ exports.fillTemplates = function (ctx, image, params, blocks, paramIgnoreMap) {
       var newContent = (dot.template(toCopy))(templateContext);
       var targetFile = getPath(paramValues, base);
       var oldContent = fs.existsSync(targetFile) ? fs.readFileSync(targetFile) : undefined;
+   
       if (!oldContent || oldContent != newContent) {
         fs.writeFileSync(targetFile, newContent);
-        logFile(file, oldContent ? "NEW".yellow : "CHANGED".green, key);
+	      logFile(file, oldContent ? "NEW".yellow : "CHANGED".green, key);
```
Comment (rhuss) [inline fp/template-engine.js]: Can we please avoid the tab here ?

File: fp/template-engine.js [hunk 2]
```diff
@@ -94,13 +94,18 @@ exports.fillTemplates = function (ctx, image, params, blocks, paramIgnoreMap) {
     var file = path + "/" + templateFile;
     var context = getTemplateContext(paramValues);
     var newContent = template(context).trim() + "\n";
+
+    var permission = getPermission(ctx.root + '/templates/' + templateFile); 
     if (!newContent.length) {
       logFile(templateFile, "SKIPPED".grey);
     } else {
-      var exists = fs.existsSync(file);
+      var exists = fs.existsSync(file);
```
Comment (rhuss) [inline fp/template-engine.js]: please remove trailing space

File: fp/template-engine.js [hunk 3]
```diff
@@ -180,6 +190,38 @@ exports.fillTemplates = function (ctx, image, params, blocks, paramIgnoreMap) {
...
+
+    var stats = fs.statSync(file);
+    return '0' + (stats.mode & parseInt('777', 8)).toString(8);
+  }
+
+  function setPermission(file, permission) {
+    try {
+     fs.chmodSync(file, permission);
+    }
+    catch(e) {}
```
Comment (rhuss) [inline fp/template-engine.js]: Shouldn't we stop here, as when the permission cant be set, then the output is not as expected ? (so not reproducible)

**Review by rhuss — COMMENTED** [review]
Thanks ! Looks good to me, with some minor nit picks.

File: fp/template-engine.js (hunk 2 above)
Comment (spaneni) [reply fp/template-engine.js]: Yes, i removed it

File: fp/template-engine.js (hunk 1 above)
Comment (spaneni) [reply fp/template-engine.js]: Yes, i removed the tab

File: fp/template-engine.js (hunk 3 above)
Comment (spaneni) [reply fp/template-engine.js]: I shift the chmodSync inside the fillTemplate method

**Review by rhuss — APPROVED** [review]
Thanks, looks good !

**Comment by rhuss** [comment]
I'm just trying to find out why the Docker automated build is failing. Looks like an issue with nodegit dependency compilation of the ssl libraries. Hmm.

**Comment by rhuss** [comment]
@spaneni could you please rebase on master, the test should succeed now.

**Comment by rhuss** [comment]
Thanks !

**Pull request merged**